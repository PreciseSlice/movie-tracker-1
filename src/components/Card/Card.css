.card {
  flex-shrink: 0;
  border-radius: 3px;
  margin: 20px 50px;
  width: 260px;
  background-size: cover;
  height: 400px; }

.movie-title {
  height: 65px;
  font-size: 1.7em;
  width: 314px;
  margin: 0 auto; }

.favorite-btn {
  background-image: url("../../assets/heart-outline.svg");
  background-repeat: no-repeat;
  border: none;
  width: 30px;
  height: 30px; }
  .favorite-btn:hover {
    background-image: url("../../assets/heart-filled.svg");
    width: 30px;
    height: 30px;
    background-size: cover; }

.favorite {
  background-image: url("../../assets/heart-filled.svg");
  width: 30px;
  height: 30px;
  background-size: cover; }

#not-signed-in {
  display: flex;
  justify-content: center;
  align-items: center;
  color: #fff;
  border-radius: 3px;
  background-color: #9c7e92;
  width: 260px;
  height: 43px;
  position: absolute;
  bottom: 0px;
  margin: 0px 0 0 50px;
  z-index: 1; }
