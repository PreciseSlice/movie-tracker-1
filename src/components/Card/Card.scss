.card {
  flex-shrink: 0;
  border-radius: 3px;
  margin: 20px 50px;
  width: 260px;
  background-size: cover;
  height: 400px;
}

.movie-title {
  height: 65px;
}

.favorite {
  background-image: url('../../assets/heart-outline.svg');
  background-repeat: no-repeat;
  border: none;
  width: 30px;
  height: 30px;
  &:hover {
    background-image: url('../../assets/heart-filled.svg');
    width: 30px;
    height: 30px;
    background-size: cover;
  }
}