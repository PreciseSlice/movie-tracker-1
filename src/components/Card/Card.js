import React, { Component } from 'react';
import './Card.css';
import PropTypes from 'prop-types';

let message;

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      clicked: false
    };
  }

  displaySignUp = loggedIn => {
    const { toggleFavorite, movieData } = this.props;
    loggedIn && toggleFavorite(movieData);
    message = !loggedIn ? <div id="not-signed-in">Sign up or log in to add favorites</div> : <div />;
    !loggedIn && this.setState({ clicked: true });
  };

  render() {
    const { movieData, loggedIn } = this.props;
    // eslint-disable-next-line
    const { title, overview, poster_path } = movieData;
    const favClass = movieData.favorite ? 'favorite' : '';
    return (
      <div>
        <h1 className="movie-title">{title}</h1>
        <article
          className="card"
          style={{ backgroundImage: `url(${poster_path})` }}
        />
        {!loggedIn && message}
        <button
          onClick={() => this.displaySignUp(loggedIn)}
          className={`favorite-btn ${favClass}`}
        />
      </div>
    );
  }
}

Card.propTypes = {
  movieData: PropTypes.object.isRequired,
  toggleFavorite: PropTypes.func,
  loggedIn: PropTypes.bool,
};

export default Card;
