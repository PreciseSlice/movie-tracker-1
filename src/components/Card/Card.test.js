import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Card from './Card';

Enzyme.configure({ adapter: new Adapter() });

describe('Card', () => {
    const  mockMovieData = {title: "Coco", 
                            id: 354912, 
                            poster: "https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg", 
                            overview: "Despite his family’s baffling generations-old ban …ck the real story behind Miguel's family history."
                           };

    it('matches the snapshot', () => {
        const wrapper = shallow(<Card movieData={mockMovieData} key={mockMovieData.id} />);
        
        expect(wrapper).toMatchSnapshot();
    });
    
});