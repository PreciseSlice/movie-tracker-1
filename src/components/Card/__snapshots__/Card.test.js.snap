// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<div>
  <h1
    className="movie-title"
  >
    Coco
  </h1>
  <article
    className="card"
    style={
      Object {
        "backgroundImage": "url(https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg)",
      }
    }
  />
  <button
    className="favorite"
  />
</div>
`;
